subprojects/scala/src/main/java/org/gradle/language/scala/tasks/BaseScalaCompileOptions.java

```diff
@@ -225,4 +231,27 @@ public IncrementalCompileOptions getIncrementalOptions() {
     public void setIncrementalOptions(IncrementalCompileOptions incrementalOptions) {
         this.incrementalOptions = incrementalOptions;
     }
+
+    /**
+     * Keeps Scala compiler daemon alive across builds for faster build times
+     * Legal values:
+     * - false (compiler is kept alive for a session)
+     * - true (compiler is kept alive across builds )
+     */
+    @Input
+    public Property<Boolean> isKeepAliveAcrossBuilds() {
```

Comment:
I don't think this has a way to toggle the property, as a new instance will be created, so something like this will have no effect:
```
 isKeepAliveAcrossBuilds().set(true) 
```

For this to work, the property needs to be assigned an instance returned by the object factory in the constructor, and we might want to make it a a getter at this point as it's strange that a method that starts with `is` does not return a boolean. 

Alternatively (and this is my interpretation of the previous comment) we could make it a `Property<KeepAliveMode>` instead of a `Boolean`. This seems to be a public API class the Gradle team is comfortable exposing to buildscripts. This way, if a new keep alive mode were to be introduced we would allready have a way to use it.

Reply:
PS: Should have started with this: I'm super grateful for this PR, Thank You!